analyzer-core/src/main/java/ffb/analyzer/core/pubsub/Publisher.java

```diff
@@ -0,0 +1,79 @@
+package ffb.analyzer.core.pubsub;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class Publisher {
+    private Object mutex = new Object();
+
+    public enum Event {
```

Comment:
I think `Event` should be an interface that we can inherit from rather than an `Enum`. This is going to be hard to maintain as we extend which events we want to handle currently; any time we want to add a new event, we have to update this enum.

The `Event` interface could expose a `runEvent`(open to changing the name) method that takes in the subscriber. At the end of the `runEvent` method, call a `notifySubscriber` method. Then, we only need to write that code once for each type of event, and it's easier to extend any way we want.